### ISSUE TITLE
mesh.prune() fails on meshes of lower dimension than space

### ISSUE THREAD

**Issue by dominik-kern** [nschloe/meshio#959]
MESHIO is a great tool, I use it particularly to convert meshes from gmsh to vtu for use in OpenGeoSys.
The extraction of boundary meshes (to apply BCs) basically works, but I run into trouble using them, because of orphaned points (in addition it is a waste of memory). The prune function, which works well for subdomains, fails on boundaries.
The 2D example 
```
import meshio

mesh = meshio.read('poisson_subdomain.msh') 	# from https://github.com/iitrabhi/dolfinx/tree/iitrabhi/mvc-xdmf/python/demo
points, cells_dict = mesh.points, mesh.cells_dict
cell_data_dict, field_data = mesh.cell_data_dict, mesh.field_data

boundary_cells = mesh.cells_dict['line']
boundary_mesh = meshio.Mesh(points=points, cells=[('line', boundary_cells)])
print(boundary_mesh)
boundary_mesh.prune() 
```
results in 

*File "/usr/local/lib/python3.8/dist-packages/meshio/_mesh.py", line 97, in prune
    all_cells_flat = numpy.concatenate([c.data for c in self.cells]).flatten()
  File "<__array_function__ internals>", line 5, in concatenate
ValueError: need at least one array to concatenate*

My humble workaround looks like 

```
 def line_mesh_prune(input_points, input_cells):
     original_shape = input_cells.shape  
     old_point_numbers = input_cells.flatten()  
     unique_point_numbers, unique_inverse = numpy.unique(old_point_numbers, return_inverse=True)
     new_points = input_points[unique_point_numbers]  
     new_cells = unique_inverse.reshape(original_shape)  
     return new_points, new_cells
```
but this functionality may be general interest.

**Comment by nschloe** [comment]
I think you're using an outdated version. Try upgrading plus
```
boundary_mesh.remove_orphaned_nodes()
```

**Comment by dominik-kern** [comment]
Thank you for your quick answer.

I upgraded from meshio 4.0.16 to 4.3.3 and, as anticipated, ``remove_orphaned_nodes()`` does the job.

Meshio gains in importance, as a growing number of users and software relies on it. Again thank you for this great tool! 
For sake of backward compatibility I would appreciate, if a function like ``prune()`` does not abruptly disappear, but give a message like "``prune()`` was removed in version X.Y, use ``remove_orphaned_nodes()`` or `remove_lower_dimensional_cells()``" instead, or even if there would be a transition like "``prune()`` is obsolete and will be removed soon".  And I hope the new functions will last for a long period.
